src/nunit.analyzers/ConstraintUsage/EqualToConstraintUsageAnalyzer.cs

```diff
@@ -0,0 +1,140 @@
+            category: Categories.Assertion,
+            defaultSeverity: DiagnosticSeverity.Info,
+            description: EqualToConstraintUsageConstants.IsNotEqualToDescription);
+
+        private static readonly string[] SupportedPositiveAssertMethods = new[] {
+            NunitFrameworkConstants.NameOfAssertThat,
+            NunitFrameworkConstants.NameOfAssertTrue,
+            NunitFrameworkConstants.NameOfAssertIsTrue
+        };
+
+        private static readonly string[] SupportedNegativeAssertMethods = new[] {
+            NunitFrameworkConstants.NameOfAssertFalse,
+            NunitFrameworkConstants.NameOfAssertIsFalse
+        };
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+            equalToDescriptor, notEqualToDescriptor);
+
+
+        protected override void AnalyzeAssertInvocation(SyntaxNodeAnalysisContext context, InvocationExpressionSyntax assertExpression, IMethodSymbol methodSymbol)
+        {
+            var negated = false;
+
+            if (SupportedNegativeAssertMethods.Contains(methodSymbol.Name))
+            {
+                negated = !negated;
+            }
+            else if (!SupportedPositiveAssertMethods.Contains(methodSymbol.Name))
+            {
+                return;
+            }
+
+            var constraint = assertExpression.GetArgumentExpression(methodSymbol, NunitFrameworkConstants.NameOfExpressionParameter);
+
+            // Constraint should be either absent, or Is.True, or Is.False
+            if (constraint != null)
+            {
+                if (!(constraint is MemberAccessExpressionSyntax memberAccess
+                       && memberAccess.Expression is IdentifierNameSyntax identifierSyntax
+                       && identifierSyntax.Identifier.Text == NunitFrameworkConstants.NameOfIs
+                       && (memberAccess.Name.Identifier.Text == NunitFrameworkConstants.NameOfIsTrue
+                           || memberAccess.Name.Identifier.Text == NunitFrameworkConstants.NameOfIsFalse)))
+                {
+                    return;
+                }
+                else if (memberAccess.Name.Identifier.Text == NunitFrameworkConstants.NameOfIsFalse)
+                {
+                    negated = !negated;
+                }
+            }
+
+            var actual = assertExpression.GetArgumentExpression(methodSymbol, NunitFrameworkConstants.NameOfActualParameter)
+                ?? assertExpression.GetArgumentExpression(methodSymbol, NunitFrameworkConstants.NameOfConditionParameter);
+
+            var shouldReport = false;
+
+            if (actual.IsKind(SyntaxKind.EqualsExpression)
+                || IsStaticObjectEquals(actual, context)
+                || IsInstanceObjectEquals(actual, context))
+            {
+                shouldReport = true;
+            }
+            else if (actual.IsKind(SyntaxKind.NotEqualsExpression)
```

Comment:
I think extracting some of the expressions into variables and/or perhaps adding "()" would help a lot on the readability.